pkger/parser.go

```diff
@@ -1434,6 +1434,54 @@ func parseChart(r Resource) (chart, []validationErr) {
 	return c, nil
 }
 
+// isDomainName is composed of 98% of the std lib implementation, only
+// difference in ours is that we only allow lower case letters.
+func isDomainName(s string) bool {
+	l := len(s)
+	if l == 0 || l > 254 || l == 254 && s[l-1] != '.' {
+		return false
+	}
+
+	last := byte('.')
+	nonNumeric := false // true once we've seen a letter or hyphen
```

Comment:
nit: this is a bit of a confusing variable name - what effect does it being nonNumeric have on this algorithm? 

>resource names can consist only of lowercase letters, numbers, dots (.), underscores (_), and hyphens (-).

given that description, i'm a little confused on what it means to be non numeric. it sounds like it's a proxy for "isDomainName." does it have to do with it not being an ip address? as in, we need a character for it to be a valid dns name, otherwise it's an ip address? 

is there some subtlety i'm missing? i feel like this could be more clearly accomplished with `isValidDomainName`.

the reason i bring it up is because this bit of code is quite intense and the clearer it its, the fewer edge cases we'll miss

Reply:
I copy pastaed this from the go std lib :-)

Reply:
its going away